moveit_ros/planning/planning_scene_monitor/src/trajectory_monitor.cpp

```diff
@@ -44,7 +45,16 @@ static const rclcpp::Logger LOGGER = rclcpp::get_logger("moveit_ros.planning_sce
 
 planning_scene_monitor::TrajectoryMonitor::TrajectoryMonitor(const CurrentStateMonitorConstPtr& state_monitor,
                                                              double sampling_frequency)
+  : TrajectoryMonitor(state_monitor, std::make_unique<TrajectoryMonitorMiddlewareHandle>(sampling_frequency),
```

Comment:
Can we use uniform initialization?

Reply:
MoveIt does not have a fixed policy on uniform initialization at the moment. I know @rhaschke does not like it very much. I on the other hand am quite a big fan.
It's almost only used in the `pilz_industrial_planner` sources (which are currently `COLCON_IGNORE`'ed in moveit2 afaik)
and while I would love to see a clang-tidy PR for the whole code base :sunglasses:, let's keep the style consistent unless it actually produces cleaner code to use it (e.g. in-class initialization of floating points, etc.).

Reply:
Arguably, uniform initialization actually solves a problem and is not a manner of style.
While C++ has multiple ways to initialize, the rules can be funny https://isocpp.org/wiki/faq/cpp11-language#uniform-init
And then of course the "most vexing parse" is always avoided.
Personally, I think within a class, the same initialization technique should be used for uniformity (unless of course you're initialing a `std::vector` or some such class which has specific initialization semantics for the parens or braces) but I'm not sure why updating to modern idioms shouldn't happen on a class-by-class basis for a codebase. If moveit originally used C-style arrays (such as amcl), and then was ported to C++, I'm not sure why one wouldn't start using `std::array` or `std::vector`.

Reply:
> I'm not sure why updating to modern idioms shouldn't happen on a class-by-class basis for a codebase.
:+1: 

It would be really nice if we could clang-tidy this change to the whole codebase but doing it in each piece as we touch it also seems fine to me.  Leaving a piece of code a bit nicer because you were there seems like an ok thing to do.

Reply:
> Arguably, uniform initialization actually solves a problem and is not a manner of style. [...] the rules can be funny

Absolutely! It's just that the places you proposed it for are not among the funny rules. And especially for those regular cases opinions differ on what is nicer. Because the whole codebase does not use uniform initialization yet, it makes sense to discuss this separately (maybe add it as a point to the next maintainer meeting and file a separate issue).

--- offtopic ---
> actually solves a problem

<details><summary>Currently my favorite problem it *creates* is this</summary>

```
$ cat > foo.cpp <<EOF
#include <iostream>

struct A {
        A(const std::initializer_list<double>& x, bool critical= true){
                std::cout << critical << "\n";
        }
};

int main(){
   A a1({0,1,2}, false);
   A a2{ {0,1,2}, false };
   A a3({0}, false);
   A a4{ {0}, false };
        return 0;
}
EOF
make foo
./foo
0
0
0
1
```
</details>

Reply:
Ah, the fun initializer_list constructor parameter.  It can both make the creating of objects more intuitive and create a trap for users.

Reply:
Ah, I see https://godbolt.org/z/GYYqWjoxv
because it's casting the boolean to a double in the initializer list.

A different, but also frustrating example
https://godbolt.org/z/orz8oT4js

Reply:
> A different, but also frustrating example

Yeah, but at least that's the "classic" one everyone knows about (or at least should after reading Effective Modern C++) :)